packages/story-editor/src/app/story/effects/useHistoryEntry.js

```diff
@@ -103,6 +104,16 @@ function useHistoryEntry({ story, current, pages, selection, capabilities }) {
         skipAddingEntry =
           JSON.stringify(adjustedPages) === JSON.stringify(adjustedEntryPages);
       }
+
+      // skip entries that have a blob url
+      // https://github.com/GoogleForCreators/web-stories-wp/issues/10289
+      pages.map((page) => {
+        page.elements.forEach((element) => {
+          if (element?.resource?.src.includes('blob:')) {
```

Comment:
Note that there's a reusable `isBlobURL` helper in the `media` package that can be used for this.